modules/developer-cli-odo-creating-a-java-application-by-using-a-devfile-in-a-disconnected-cluster.adoc

```diff
@@ -0,0 +1,193 @@
+        - name: m2
+          path: /home/user/.m2
+  - name: m2
+    volume: {}
+commands:
+  - id: mvn-package
+    exec:
+      component: tools
+      commandLine: "mvn -Dmaven.repo.local=/home/user/.m2/repository -Dhttp.proxyHost=<proxy-host> -Dhttp.proxyPort=<proxy-port> -Dhttps.proxyHost=<proxy-host> -Dhttps.proxyPort=<proxy-port> package"
+      group:
+        kind: build
+        isDefault: true
+  - id: run 
+    exec:
+      component: tools
+      commandLine: "java -jar target/*.jar"
+      group:
+        kind: run
+        isDefault: true
+  - id: debug
+    exec:
+      component: tools
+      commandLine: "java -Xdebug -Xrunjdwp:server=y,transport=dt_socket,address=${DEBUG_PORT},suspend=n -jar target/*.jar"
+      group:
+        kind: debug
+        isDefault: true
+----
+
+. Create a Java application:
++
+[source,terminal]
+----
+$ odo create java-maven --devfile <path-to-your-devfile> --starter
+----
++
+.Example output
+[source,terminal]
+----
+Validation
+ ✓  Checking devfile existence [87716ns]
+ ✓  Creating a devfile component from registry: DefaultDevfileRegistry [107247ns]
+ ✓  Validating devfile component [396971ns]
+ 
+ Starter Project
+ ✓  Downloading starter project springbootproject from https://github.com/odo-devfiles/springboot-ex.git [2s]
+
+Please use `odo push` command to create the component with source deployed
+----
+
+. Push the changes to the cluster:
++
+[source,terminal]
+----
+$ odo push
+----
++
+.Example output
+[source,terminal]
+----
+I0224 14:43:18.802512   34741 util.go:727] HTTPGetRequest: https://raw.githubusercontent.com/openshift/odo/master/build/VERSION
+I0224 14:43:18.833631   34741 context.go:115] absolute devfile path: '/Users/pkumari/go/src/github.com/openshift/odo/testim/devfile.yaml'
+[...]
+Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/3.2.1/plexus-utils-3.2.1.jar (262 kB at 813 kB/s)
+[INFO] Replacing main artifact with repackaged archive
+[INFO] ------------------------------------------------------------------------
+[INFO] BUILD SUCCESS
+[INFO] ------------------------------------------------------------------------
+[INFO] Total time:  19.638 s
+[INFO] Finished at: 2021-02-24T08:59:30Z
+[INFO] ------------------------------------------------------------------------
+ ✓  Executing mvn-package command "mvn -Dmaven.repo.local=/home/user/.m2/repository -Dhttp.proxyHost=<proxy-host> -Dhttp.proxyPort=<proxy-port> -Dhttps.proxyHost=<proxy-host> -Dhttps.proxyPort=<proxy-port> package" [23s]
+ •  Executing run command "java -jar target/*.jar"  ...
+I0224 14:29:30.557676   34426 exec.go:27] Executing command [/opt/odo/bin/supervisord ctl start devrun] for pod: java-maven-5b8f99fcdb-9dnk6 in container: tools
+devrun: started
+ ✓  Executing run command "java -jar target/*.jar" [3s]
+
+Pushing devfile component java-maven
+ ✓  Changes successfully pushed to component
+----
+
+. Verify that the application has stared by displaying logs:
```

Comment:
```suggestion
. Verify that the application has started by displaying logs:
```

Reply:
@rolfedh what I meant to say here was "Display the logs to verify that the application has started" - WDYT of this wording?

Reply:
That's even better.